rl_loop/update_resign_threshold.py

```diff
@@ -61,22 +69,32 @@ def update_flagfile(flags_path, new_threshold):
     old_threshold = RESIGN_FLAG_REGEX.search(lines).groups(1)
     lines = re.sub(RESIGN_FLAG_REGEX, "--resign_threshold={:.3f}".format(new_threshold), lines)
 
-    print("Updated percentile from {} to {:.3f}".format(old_threshold, new_threshold))
-    with tf.gfile.GFile(flags_path, 'w') as f:
-        f.write(lines)
+    if abs(float(old_threshold[0]) - new_threshold) < 0.001:
+        print("Not updating percentiles; {} ~= {:.3f}".format(
+                old_threshold[0], new_threshold), flush=True)
+    else:
+        print("Updated percentile from {} to {:.3f}".format(
+                old_threshold[0], new_threshold), flush=True)
+        with tf.gfile.GFile(flags_path, 'w') as f:
+            f.write(lines)
 
 
 def main(argv):
     if len(argv) > 1:
         raise app.UsageError('Too many command-line arguments.')
 
-    if not 'CBT_TABLE' in os.environ:
-        raise app.UsageError('CBT_TABLE not set')
+    games_nr = bigtable_input.GameQueue(
+        FLAGS.cbt_project, FLAGS.cbt_instance, FLAGS.cbt_table + '-nr')
 
     while True:
-        new_pct = get_95_percentile_bleak()
+        new_pct = get_95_percentile_bleak(games_nr)
         update_flagfile(fsdb.flags_path(), new_pct)
         time.sleep(60 * 3)
 
 if __name__ == '__main__':
-    app.run(main)
+    valid_flags = list(map(lambda f: '--' + f, FLAGS.flag_values_dict().keys()))
```

Comment:
so this jumble here happened because this code directly imports bigtable_input, which isn't part of the rl_loop set of modules. So it's crossing a boundary that's usually bridged via masked_flags.run and subprocesses. If you move the bigtable code here into its own top level module get_resign_rate.py, then use masked_flags.run(), then that'll clean stuff up.

Reply:
we should probably put a README in rl_loop documenting the boundary's expectations